### PR TITLE
[EA Forum only] update virtual program dates

### PR DESCRIPTION
**Pull request by s-cheng**
Updates the VP deadline and start dates to match what's [here](https://www.effectivealtruism.org/virtual-programs), also update the logic to more closely match what the program uses for their dates.

Screen Shot 2023-01-07 at 11 04 23 AM

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203667707273338) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/events/modules/VirtualProgramCard.tsx [hunk 1]
```diff
@@ -104,30 +104,17 @@ const VirtualProgramCard = ({program, classes}: {
...
-  const jan1 = moment("01-01-2023", "MM-DD-YYYY")
-  const jan29 = moment("01-29-2023", "MM-DD-YYYY") // deadline for Feb-Mar cohort? (should be normal start date logic)
-  if (moment().isSameOrBefore(dec11)) {
-    deadline = dec11
-  } else if (moment().isSameOrBefore(jan1)) {
-    deadline = jan29
+  // Find the next deadline for applying to the Intro VP, which is usually the 4th Sunday of every month
+  // (though it will sometimes move to the 5th Sunday - this is not accounted for in the code).
+  // This defaults to the Sunday in the week of the 28th day of this month.
+  let deadline = moment().date(28).day(0)
```
Comment (jpaddison3) [inline packages/lesswrong/components/events/modules/VirtualProgramCard.tsx:110]: That is a stupid name for the dayOfWeek function

File: packages/lesswrong/components/events/modules/VirtualProgramCard.tsx [hunk 2]
```diff
@@ -104,30 +104,17 @@ const VirtualProgramCard = ({program, classes}: {
...
+  // If that Sunday is in the past, use next month's 4th Sunday.
+  if (deadline.isBefore(moment())) {
+    deadline = moment().add(1, 'months').date(28).day(0)
```
Comment (jpaddison3) [inline packages/lesswrong/components/events/modules/VirtualProgramCard.tsx:113]: What happens if:
* Today is the 31st
* I add one month
* Next month has <31 days
Does it round down to the max day of the next month, or roll over to the following month?

**Review by jpaddison3 — APPROVED** [review]
ASTLAC, TY

File: packages/lesswrong/components/events/modules/VirtualProgramCard.tsx (hunk 2 above)
Comment (s-cheng) [reply packages/lesswrong/components/events/modules/VirtualProgramCard.tsx:113]: Ah good question, I didn't test that case. According to the [docs](https://momentjs.com/docs/#/manipulating/):

> If the day of the month on the original date is greater than the number of days in the final month, the day of the month will change to the last day in the final month.

Which is what we want, so it should be good.

File: packages/lesswrong/components/events/modules/VirtualProgramCard.tsx (hunk 1 above)
Comment (s-cheng) [reply packages/lesswrong/components/events/modules/VirtualProgramCard.tsx:110]: Agree

**Pull request merged**